src/BayesFilters/src/GaussianMixture.cpp

```diff
@@ -41,7 +41,42 @@ GaussianMixture::GaussianMixture
 }
 
 
-GaussianMixture::~GaussianMixture() noexcept { }
+GaussianMixture::~GaussianMixture() noexcept
+{ }
+
+
+void GaussianMixture::resize(const std::size_t components, const std::size_t dim_linear, const std::size_t dim_circular)
+{
+    std::size_t new_dim = dim_linear + dim_circular;
+
+    if ((this->dim_linear == new_dim) && (this->dim_circular == dim_circular) && (this->components == components))
+        return;
+    else if ((this->dim == new_dim) && (this->components == components))
```

Comment:
This if can removed as both `dim` and `components` are correct and `dim_linear` and `dim_circular` are updated at the end of the method.